.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+name: Node.js CI
+
+on:
+  pull_request:
+    branches: [ "develop" ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '18.x'
+        cache: 'npm'
+    - run: npm install nx
+    - run: npm install
+    - run: npm run lint
+    
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '18.x'
+        cache: 'npm'
+    - run: npm install nx
+    - run: npm install
+    - run: npm run test
+
+  type-check:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '18.x'
+        cache: 'npm'
+    - run: npm install nx
+    - run: npm install
+    - run: cd apps/client && tsc --noEmit --esModuleInterop && cd ../server && tsc --noEmit --esModuleInterop
```

Comment:
You can run this action with either `nx` cli or add npm scripts to the top level.